### PR TITLE
Wait for elasticsearch/opensearch to be ready before running the integration tests

### PR DESCRIPTION
**Pull request by Ashmita152**
Signed-off-by: Ashmita Bohara <ashmita.bohara@logz.io>

## Which problem is this PR solving?
- Resolves https://github.com/jaegertracing/jaeger/issues/3411

## Short description of the changes
- We are waiting for elasticsearch/opensearch to be ready before running the integration tests
- Bumping opensearch container version to 1.2.0.

### CODE REVIEW

**Review by pavolloffay — APPROVED** [review]

File: scripts/es-integration-test.sh [hunk 1]
```diff
@@ -58,6 +58,23 @@ setup_query() {
...
+    --write-out
+    ''%{http_code}''
+  )
+  local counter=0
+  while [[ "$(curl ${params[@]} ${url})" != "200" && ${counter} < 30 ]]; do
+    sleep 2
+    counter=$((counter+1))
+    echo "waiting for ${url} to be up..."
+  done
+}
```
Comment (yurishkuro) [inline scripts/es-integration-test.sh:76]: I would recommend failing the whole script if 200 was never achieved, this way we'd get a much earlier and clearer failure signal.

Comment (yurishkuro) [reply scripts/es-integration-test.sh:76]: i.e. something like this:
```
while counter < 30 {
    status = $(curl)
    if status == 200 then return 0
}
return 1
```

**Review by yurishkuro — CHANGES_REQUESTED** [review]
per comments

**Comment by Ashmita152** [comment]
Hi Yuri, sorry for the huge delay on my side.

**Review by yurishkuro — APPROVED** [review]
👏 👏 👏

**Pull request merged**